src/components/SidePanel/SidePanel.tsx

```diff
@@ -108,20 +113,39 @@ const SidePanelComponent = ({
   pinned,
   width = "",
   parentId = "l-application",
+  show,
 }: Props): React.JSX.Element => {
+  const [hiding, setHiding] = useState(true);
+  const [previousShow, setPreviousShow] = useState(false);
+
   const container = document.getElementById(parentId) || document.body;
 
+  const animated = show !== undefined;
+  show ??= true;
+
+  if (show !== previousShow) {
+    setPreviousShow(show);
+    setHiding(true);
+  }
+
+  if (!show && !hiding) {
+    return null;
+  }
+
   return (
     <>
       {createPortal(
         <AppAside
           className={classnames("side-panel", className, {
             "is-overlay": overlay,
+            "slide-in": animated,
           })}
+          collapsed={!show}
```

Comment:
One issue with using only this is that if you have many side panels on a page that can be opened, they will all be in the DOM even though all are closed. I think the approach in `ToastNotification` is a little better of wrapping the rendered things in `Animate`, which in turn removes them from the DOM if they are not shown. This ensures the DOM is clean of (possibly many) side panels that are not opened.

Reply:
I checked the demo server previously, and maybe I was just unlucky to pick the pinned story](https://react-components-1243.demos.haus/?path=/story/components-sidepanel--pinned). When inspecting the DOM for the panel and closing it, it stays in the DOM.

Though you are right, all the other stories do unmount it from the DOM, so this concern is probably to be resolved. Well, at least reduced to the question of why the pinned story does not unmount.